__tests__/account.outsideExecution.test.ts

```diff
@@ -1,119 +1,344 @@
-      { maxFee: 1e18 }
-    );
+    // Deploy the SNIP-9 signer account (ArgentX v 0.4.0, using SNIP-9 v2):
+    const calldataAX = new CallData(compiledArgentX4Account.abi);
+    const axSigner = new CairoCustomEnum({ Starknet: { pubkey: targetPubK } });
+    const axGuardian = new CairoOption<unknown>(CairoOptionVariant.None);
+    const constructorAXCallData = calldataAX.compile('constructor', {
+      owner: axSigner,
+      guardian: axGuardian,
+    });
+    const response = await executorAccount.declareAndDeploy({
+      contract: compiledArgentX4Account,
+      classHash: '0x36078334509b514626504edc9fb252328d1a240e4e948bef8d0c08dff45927f',
+      compiledClassHash: '0x7a663375245780bd307f56fde688e33e5c260ab02b76741a57711c5b60d47f6',
+      constructorCalldata: constructorAXCallData,
+    });
     const targetAddress = response.deploy.contract_address;
-    target = new Account(provider, targetAddress, targetPK);
+    signerAccount = new Account(provider, targetAddress, targetPK);
 
-    // Transfer some tokens to the target account
+    // Transfer dust of ETH token to the signer account
     const transferCall = {
-      contractAddress: devnetERC20Address,
+      contractAddress: ethAddress,
       entrypoint: 'transfer',
       calldata: {
-        recipient: targetAddress,
-        amount: uint256(transferAmount),
+        recipient: signerAccount.address,
+        amount: cairo.uint256(1000),
       },
     };
-
-    const { transaction_hash } = await sender.execute(transferCall, undefined, { maxFee: 1e18 });
+    const { transaction_hash } = await executorAccount.execute(transferCall);
     await provider.waitForTransaction(transaction_hash);
   });
 
-  it('target account should support SNIP-9', async () => {
-    const res = await sender.callContract({
-      contractAddress: target.address,
-      entrypoint: 'supports_interface',
-      calldata: [SNIP9_V1_INTERFACE_ID],
+  test('getOutsideCall', async () => {
+    const call1: Call = {
+      contractAddress: '0x0123',
+      entrypoint: 'transfer',
+      calldata: {
+        recipient: '0xabcd',
+        amount: cairo.uint256(10),
+      },
+    };
+    expect(outsideExecution.getOutsideCall(call1)).toEqual({
+      to: '0x0123',
+      selector: getSelectorFromName(call1.entrypoint),
+      calldata: ['43981', '10', '0'],
     });
+  });
 
-    expect(res[0]).toBe('0x1');
+  test('Build SNIP-9 v2 TypedData', async () => {
+    const call1: Call = {
+      contractAddress: '0x0123',
+      entrypoint: 'transfer',
+      calldata: {
+        recipient: '0xabcd',
+        amount: cairo.uint256(10),
+      },
+    };
+    const callOptions: OutsideExecutionOptions = {
+      caller: '0x1234',
+      execute_after: 100,
+      execute_before: 200,
+    };
+    const message: TypedData = outsideExecution.getTypedData(
+      constants.StarknetChainId.SN_SEPOLIA,
+      callOptions,
+      21,
+      [call1],
+      OutsideExecutionVersion.V2
+    );
+    expect(message).toEqual({
```

Comment:
Don't change anything now but just wanted to mention that in the future when dealing with large objects as the expected result of tests using snapshots might be simpler.